guides/source/action_mailbox_basics.md

```diff
@@ -272,8 +272,8 @@ Configure basic routing:
 ```ruby
 # app/mailboxes/application_mailbox.rb
 class ApplicationMailbox < ActionMailbox::Base
-  routing /^save@/i     => :forwards
-  routing /@replies\./i => :replies
+  routing (/^save@/i     => :forwards)
+  routing (/@replies\./i => :replies)
```

Comment:
I _think_ this is a syntax error because of the space between `routing` and `(`.

If I'm correct, I'm surprised it isn't caught by RuboCop, especially given https://github.com/rails/rails/pull/47186#discussion_r1091430261... 🤔

Was this autocorrected?

Reply:
@sambostock Thanks for taking a look!

This is not a syntax error, although I would prefer `routing(...)`, it doesn't seem like there is a rule for that at least in our config.

```ruby
>> RUBY_VERSION
=> "3.3.0"
?> def foo bar
?>   puts bar
>> end
=> :foo
>>
>> foo "omg"
omg
=> nil
>> foo("omg")
omg
=> nil
>> foo ("omg")
omg
=> nil
```

Reply:
I think the syntax error that @sambostock tells is as follows:

## With space

invalid syntax:

```console
% ruby -vce 'routing (/^save@/i     => :forwards)'
ruby 3.2.0 (2022-12-25 revision a528908271) [x86_64-darwin19]
-e:1: void value expression
...ing (/^save@/i     => :forwards)
-e: compile error (SyntaxError)
```

## No space

valid syntax:

```console
% ruby -vce 'routing(/^save@/i     => :forwards)'
ruby 3.2.0 (2022-12-25 revision a528908271) [x86_64-darwin19]
Syntax OK
```

The same is true for Ruby 3.3.0-dev.

Reply:
Thanks! I will fix it.

Reply:
@sambostock @koic Fixed in 321c8df, but I wonder why `Ripper.sexp` didn't catch this? Is there another rule we can add that would have caught it?

Reply:
I'm not sure, but maybe there is something wrong with the Parser gem. Unfortunately, the cause has not been determined.

Reply:
Both Ripper and Parser recognize this syntax:

```ruby
irb(main):003:0> Ripper.sexp('routing (/^save@/i     => :forwards)')
=> 
[:program,
   [:@ident, "routing", [1, 0]],
   [:args_add_block,
    [[:paren,
      [:case,
       [:regexp_literal, [[:@tstring_content, "^save@", [1, 10]]], [:@regexp_end, "/i", [1, 16]]],
       [:in, [:symbol_literal, [:symbol, [:@ident, "forwards", [1, 27]]]], nil, nil]]]],
    false]]]]

irb(main):006:0> Parser::CurrentRuby.parse('routing (/^save@/i     => :forwards)')
=> 
s(:send, nil, :routing,                                                      
  s(:begin,                                                                  
    s(:match_pattern,                                                
      s(:regexp,                                                     
        s(:str, "^save@"),                                           
        s(:regopt, :i)),                                             
      s(:sym, :forwards)))) 
```

But:

```sh
$ ruby -w -e 'routing (/^save@/i     => :forwards)'

-e:1: void value expression
...ing (/^save@/i     => :forwards)
-e: compile error (SyntaxError)
```

🤷‍♂️